src/view/dataElements/xdmObject/helpers/parseResponseJson.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Parses the JSON returned by a network request and returns an object that looks like a network request
+ * Useful for post-fetch() operations where status and body payloads need to be evaluated at the same time
+ *
+ * @param  {object} response A response from a network request
+ *
+ * @return {object}          The parsed JSON, status from the response
+ */
+export default response => {
+  return new Promise(resolve => {
+    response.json().then(body =>
+      resolve({
+        status: response.status,
+        ok: response.ok,
+        json: () => {
+          return body;
+        }
+      })
+    );
+  });
```

Comment:
No need to create a new promise here. Try this.
  
```
return response.json().then(body => {
  return {
    status: response.status,
    ok: response.ok,
    json() {
      return body;
    }
  }
})
```

I haven't looked at the rest of this PR yet, but you'll need to make sure we're appropriately handling a `response.ok` of `false`.